### ISSUE TITLE
D5100 1080i/720p Clean HDMI aspect ratio

### ISSUE THREAD

**Issue by adammartinai** [simeonpilgrim/nikon-firmware-tools#102]
I've uploaded the hacked firmware for my D5100.

When I enable the `HDMI Output 1080i FullScreen Fixed` feature I get a black border around my 'Clean HDMI' output.

On the other hand; when I enable the `HDMI Output 720p Fullscreen` I get an unusual aspect ratio for. my 'Clean HDMI'

Am I missing something? Do I need to enable another feature? or make sure I disable a feature?

**Comment by adammartinai** [comment]
I'm using the firmware `D5100_0102.bin` with an md5 of `2214210ad2c65b5e857899ca79f3da1`

**Comment by johnruth** [comment]
I'm having the same issue with 720p.  When I update the firmware file to use 1080i fullscreen I get noticable lag/stuttering.

**Comment by drmaniac** [comment]
I have the exact same issue (borders) when I connect the D5100 with a HDMI capture device. On the other hand if I connect the camera to my TV it works like a charm.

**Comment by johnnyghostly** [comment]
> > I'm having the same issue with 720p. When I update the firmware file to use 1080i fullscreen I get noticable lag/stuttering.
> 
> Getting the same behavior about the lag/stuttering, have you been able to solve your issue?
> Someone else seems to have the same behavior according to this: https://nikonhacker.com/viewtopic.php?f=4&t=2600

I was not able to find a solution, but I found a workaround on Reddit.  Switching to PAL instead of NTSC has less lag/stuttering.

**Comment by goldyfruit** [comment]
> > > I'm having the same issue with 720p. When I update the firmware file to use 1080i fullscreen I get noticable lag/stuttering.
> > 
> > 
> > Getting the same behavior about the lag/stuttering, have you been able to solve your issue?
> > Someone else seems to have the same behavior according to this: https://nikonhacker.com/viewtopic.php?f=4&t=2600
> 
> I was not able to find a solution, but I found a workaround on Reddit. Switching to PAL instead of NTSC has less lag/stuttering.

Yeah I found the issue, for some reason **30fps** is the thing causing issue, using **24fps** doesn't cause any stuttering.
**NTSC** only allows **24fps** which is why it's working for me.

Thanks for answering.

**Comment by mil-ad** [comment]
> I have the exact same issue (borders) when I connect the D5100 with a HDMI capture device.

having exactly the same issue

**Comment by goldyfruit** [comment]
> > I have the exact same issue (borders) when I connect the D5100 with a HDMI capture device.
> 
> having exactly the same issue

You will need to apply the full-screen patch from https://simeonpilgrim.com/nikon-patch/nikon-patch.html
```
Beta - HDMI Output 1080i FullScreen Fixed
Beta - HDMI Output 720p FullScreen
```

**Comment by mil-ad** [comment]
> You will need to apply the full-screen patch from https://simeonpilgrim.com/nikon-patch/nikon-patch.html

I have, but similar to @adammartinai 720p patch gives a weird aspect ratio and the 1080i gives black borders. 

Actually I don't think the 1080i patch makes a difference for me in terms of the black box.

**Comment by goldyfruit** [comment]
> > You will need to apply the full-screen patch from https://simeonpilgrim.com/nikon-patch/nikon-patch.html
> 
> I have, but similar to @adammartinai 720p patch gives a weird aspect ratio and the 1080i gives black borders.
> 
> Actually I don't think the 1080i patch makes a difference for me in terms of the black box.

I have a D5100, I applied the `1.02` patch with `Beta - HDMI Output 1080i FullScreen Fixed`, no issue so far.

**Comment by mil-ad** [comment]
Interesting! I have the same camera and the same firmware. What other patches have you applied? Presumably the clean HDMI one?

**Comment by goldyfruit** [comment]
> Interesting! I have the same camera and the same firmware. What other patches have you applied? Presumably the clean HDMI one?

Here are the options I choosed.
![image](https://user-images.githubusercontent.com/614115/122257340-ee695580-ce9d-11eb-8fce-82e98204e663.png)

**Comment by simeonpilgrim** [comment]
some of the problem is the image sensor is not the same aspect ratio as the hdmi outputs, so has to do binning (to be able to have reduced data for IO reasons), which scales the image, and then that image is rescaled to the output image. 

The stock firmware keeps the image ratio of the input image to the output image, and thus the bars.
The first version of the hacks just altered the output resolution to be 100% thus the aspect ratio where wrong. 
To correct this in the 1080i patch extra code was injected to change the binning sizes so the ratio was 'normal'. 
But that then caused (I think I recall) output glitches. under some modes/settings. It's almost like a rather complex problem with sets of constraints.

For myself, my D5100 died, and with my D7000 I was not doing video, and by the time I wanted the clean HDMI out, I owned a D750 which just does it out of the box.. So for myself I have no hardware to test on, and no personal need.

**Comment by vasylyamnych** [comment]
@goldyfruit Thanks a lot - you saved my day!!!
I had the same issue with the black border and the only difference at firmware was that I was using `Video 1080 HQ 54mbps Bit-rate`. And when I changed to `Video 1080 HQ 64Mbps, NQ 22Mbps` then black borders disappeared.

**Comment by vasylyamnych** [comment]
@simeonpilgrim thanks a lot for your great work - your firmware brings a new life to my old camera. Now it is a webcam.

**Comment by mistrlns** [comment]
> capture

I'm running into the same issue. I think that the problem is that the 1080 is interlaced instead of progressive. So maybe the capture card doesn't support that. I tried changing the output to 720p on the TV as an experiment and it had the black borders just like on the capture card.

**Comment by mistrlns** [comment]
> 



> some of the problem is the image sensor is not the same aspect ratio as the hdmi outputs, so has to do binning (to be able to have reduced data for IO reasons), which scales the image, and then that image is rescaled to the output image.
> 
> The stock firmware keeps the image ratio of the input image to the output image, and thus the bars. The first version of the hacks just altered the output resolution to be 100% thus the aspect ratio where wrong. To correct this in the 1080i patch extra code was injected to change the binning sizes so the ratio was 'normal'. But that then caused (I think I recall) output glitches. under some modes/settings. It's almost like a rather complex problem with sets of constraints.
> 
> For myself, my D5100 died, and with my D7000 I was not doing video, and by the time I wanted the clean HDMI out, I owned a D750 which just does it out of the box.. So for myself I have no hardware to test on, and no personal need.



> I have the exact same issue (borders) when I connect the D5100 with a HDMI capture device. On the other hand if I connect the camera to my TV it works like a charm.

I'm running into the same issue. I think that the problem is that the 1080 is interlaced instead of progressive. So maybe the capture card doesn't support that. I tried changing the output to 720p on the TV as an experiment and it had the black borders just like on the capture card.

**Comment by mistrlns** [comment]
> I have the exact same issue (borders) when I connect the D5100 with a HDMI capture device. On the other hand if I connect the camera to my TV it works like a charm.

I'm running into the same issue. I think that the problem is that the 1080 is interlaced instead of progressive. So maybe the capture card doesn't support that. I tried changing the output to 720p on the TV as an experiment and it had the black borders just like on the capture card.

**Comment by Yalovega** [comment]
> @simeonpilgrimбольшое спасибо за ваш большой труд - ваша прошивка дает новую жизнь моей старой камере. Теперь это веб-камера.
I have a question for you, when you installed the firmware on pure hdmi, do you need to include something in the settings so that there is pure hdmi or something?

**Comment by ProHyphonzDaCoder** [comment]
is the stuttering fixed?

**Comment by simeonpilgrim** [comment]
nobody has done anything no. But I do now own a D5100, so could do some testing.. It seems like we needed to be more explicit about what was being altered.

**Comment by ProHyphonzDaCoder** [comment]
@simeonpilgrim  Yeah, If you could in the future, can you please make it so that the hdmi output on capture card doesnt have black borders?

**Comment by simonking** [comment]
I'll add in my experience. No matter what I do, including the settings posted by @goldyfruit, I get black bars on 1080 output. I can get 720 without black bars, but the aspect ratio is stretched horizontally. It seems like the correction from sensor to HDMI output isn't _quite_ correct. It's distorted enough that people mention it to me on video calls, so for now it's not workable for me to use my 5100 as a webcam. I've been testing different combinations of options but haven't found anything that works.

**Comment by ProHyphonzDaCoder** [comment]
> I'll add in my experience. No matter what I do, including the settings posted by @goldyfruit, I get black bars on 1080 output. I can get 720 without black bars, but the aspect ratio is stretched horizontally. It seems like the correction from sensor to HDMI output isn't _quite_ correct. It's distorted enough that people mention it to me on video calls, so for now it's not workable for me to use my 5100 as a webcam. I've been testing different combinations of options but haven't found anything that works.

Simeon mentioned about testing to improve this feature so maybe it might be suitable in a few months or years?

**Comment by simonking** [comment]
For now, I found a workaround by using [OBS Studio](https://obsproject.com/), which allows you to manually transform an incoming video signal. For my D5100 transforming the size of the incoming video to 1600x1080 resulted in a corrected aspect ratio. I keep the bounding box size at 1920x180 and choose "Scale to width of bounds" so that it crops to a full screen. 

<img width="557" alt="image" src="https://user-images.githubusercontent.com/1409075/217617576-d8999506-9c30-4ce2-b7e1-d4a7588cd876.png">

 Although it's an extra step, OBS also offers some great tools. I'm still happy to be able to use this D5100 for this purpose and the other aspects of the firmware patch (clean HDMI, Live View 6 hours) work great in combination with this.

**Comment by ProHyphonzDaCoder** [comment]
> For now, I found a workaround by using [OBS Studio](https://obsproject.com/), which allows you to manually transform an incoming video signal. For my D5100 transforming the size of the incoming video to 1600x1080 resulted in a corrected aspect ratio. I keep the bounding box size at 1920x180 and choose "Scale to width of bounds" so that it crops to a full screen. 
> 
> 
> 
> <img width="557" alt="image" src="https://user-images.githubusercontent.com/1409075/217617576-d8999506-9c30-4ce2-b7e1-d4a7588cd876.png">
> 
> 
> 
>  Although it's an extra step, OBS also offers some great tools. I'm still happy to be able to use this D5100 for this purpose and the other aspects of the firmware patch (clean HDMI, Live View 6 hours) work great in combination with this.

Any clue what live view does?

**Comment by goldyfruit** [comment]
I'm using [BlackMagic Intensity Pro 4K](https://www.blackmagicdesign.com/ca/products/intensitypro4k) as HDMI capture device, maybe it does some things which is why I don't have any trouble.

**Comment by ProHyphonzDaCoder** [comment]
> I'm using [BlackMagic Intensity Pro 4K](https://www.blackmagicdesign.com/ca/products/intensitypro4k) as HDMI capture device, maybe it does some things which is why I don't have any trouble.

Perhaps, but i use a usb stick capture card not a PCIE Capture card

**Comment by mistrlns** [comment]
The 1080i patch works when i connect my D5100 to my TV or to my video monitor but not when i plug it into the cheap BlueAVS usb capture card. You get what you pay for.

**Comment by bigbluebird1981** [comment]
Coming in way late on this, but if anyone can point me in the right direction I would appreciate it.  The online Patch Tool seems to be non functional (chrome).  I am trying to find the 5100 1.02 firmware file.  Any direction would be appreciated as it looks like I am 10 years to late to get it.

**Comment by ian5142** [comment]
You can find the factory firmware here: https://downloadcenter.nikonimglib.com/en/download/fw/20.html
After downloading that, upload it to the Nikon Patch site, to modify it.

**Comment by bigbluebird1981** [comment]
Thank you, you rock

**Comment by ABpython** [comment]
Hello! I'm writing this in 2026. I decided to take my old Nikon D5100 camera and set it up for streaming in OBS, I noticed that after patching the camera with the setting (HDMI Output 1080i FullScreen Fixed) - in OBS my image in the picture was freezing every couple of seconds.

Solution: I set PAL instead of NTSC, and I set the movie quality to 1080 (24 frames per second) - the freezing disappeared, and the camera quality did not change (or imperceptibly)

**Comment by Alekai1X** [comment]
I've downloaded the software hack but it only let's me go into Auto and some other modes. If I choose manual, I get a black screen. The other modes don't let me adjust ISO. Please help. I'm using the D5100 with an Elgato camlink.

<img width="598" height="549" alt="Image" src="https://github.com/user-attachments/assets/66382677-dc38-4673-94bf-42f577129f4f" />
